tests/console/irqbalance.pm

```diff
@@ -0,0 +1,52 @@
+# SUSE's openQA tests
+#
+# Copyright © 2009-2013 Bernhard M. Wiedemann
+# Copyright © 2012-2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Package: irqbalance
+# Summary: Test irqbalance runs as service as well as standalone
+# Maintainer: Pavel Dostál <pdostal@suse.cz>
+
+use warnings;
+use base "consoletest";
+use strict;
+use testapi qw(is_serial_terminal :DEFAULT);
+use utils qw(systemctl zypper_call clear_console);
+
+sub run {
+    my $self = shift;
+    $self->select_serial_terminal;
+
+    # Return when balancing is ineffective on system with a single cpu
+    zypper_call("in coreutils") if (script_run("which nproc") != 0);
+    if (script_output('nproc') <= 1) {
+        record_info("Low CPUs amount", "Balancing is ineffective on systems with a single cpu");
+        return;
+    }
+
+    # Install irqbalance
+    zypper_call("in irqbalance") if (script_run("which irqbalance") != 0);
+
+    # Enable and start irqbalance.service if needed
+    systemctl('start irqbalance.service') if (script_run('systemctl -n is-active irqbalance.service'));
+    systemctl('enable irqbalance.service') if (script_run('systemctl -n is-enabled irqbalance.service'));
+
+    # Check that the irqbalance.service is running
+    systemctl('is-active irqbalance.service');
+    systemctl('status irqbalance.service');
+
+    # Run the irqbalance in oneshot debug modes
+    assert_script_run("irqbalance --oneshot --debug", 360);
+
+    assert_script_run("cat /proc/interrupts");
```

Comment:
This alone does not do much. What is the purpose of this call?

Reply:
Honestly, I don't know for what to grep. It's there just for debug. I'll remove it then.

Reply:
Should we maybe check `/proc/interrupts` since it was the cause of [this regression](https://bugzilla.suse.com/show_bug.cgi?id=1178477)? Maybe helping to block a similar update in the future? Just a genuine question, since I have not followed this

Reply:
@pdostal Thanks for starting it. FYI. The purpose of poo#92815 is given in its description, main purpose is to detec the exact issue in bsc#1178477. Regarding what this issue is, the root cause of it, and how to detect it, they are provided in one of its relevant issues, https://progress.opensuse.org/issues/93249#note-2 and https://progress.opensuse.org/issues/93249#note-4.

Reply:
@pdostal Thanks for starting it. FYI. The purpose of poo#92815 is given in its description, main purpose is to detec the exact issue in bsc#1178477. Regarding what this issue is, the root cause of it, and how to detect it, they are provided in one of its relevant issues, https://progress.opensuse.org/issues/93249#note-2 and https://progress.opensuse.org/issues/93249#note-4.

Reply:
In last virtualization maintenance meeting we agreed with @alice-suse that I XEN specific checks will be added to this test module in separate pull request.